metabase/metabase
# Migrate notebook aggregations to MLv2 — Operators (1)

Part of #31001

Adds TypeScript wrappers for MLv2 aggregation functions and _partially_ migrates notebook's `AggregateStep` to MLv2.
This is the first step in this migration, and it only migrates common operators like count, average, max, min, etc

> **Warning**
> This PR targets a feature branch and doesn't migrate `AggregateStep` features 1:1
> This branch is expected to have failing tests because parts of functionality are simply missing
> You can monitor the migration health with the last branch in the chain — #31530

Also, there's a known BE issue that temporal columns are not highlighted as selected — #31555

### To Verify

For different kinds of questions (structured/native, based on raw data/models/questions, flat/multi-stage):

1. Open `/question/:id/notebook`
2. Play around with the aggregation step ("Pick the metric you want to see")
3. Try out different column types (real/computed, with/without binning, with/without temporal bucketing)
4. Overall ensure the behavior matches `master`

### Demo

https://github.com/metabase/metabase/assets/17258145/21d4d939-5e09-480a-b399-323aad377f99
